File: Python/finder/finder.py
import os
import sys
import subprocess


target = sys.argv[1]
is_dir = target.endswith('/')
directory = os.getcwd()

while True:
    if (
        (is_dir and os.path.isdir(os.path.join(directory, target))) or
        (not is_dir and os.path.isfile(os.path.join(directory, target)))
    ):
        print(directory)
        exit(0)
    if directory == '/':
        exit(1)
    directory = os.path.dirname(directory)

File: javascript/projects/finder/finder.js
import path from 'node:path';
import fs from 'node:fs';

const target = process.argv[2];

let directory = process.cwd();

while (true) {
  const exists = fs.existsSync(path.join(directory, target));
  if (exists) {
    console.log(directory);
    process.exit(0);
  }
  if (directory === path.parse(directory).root) {
    process.exit(1);
  }
  directory = path.dirname(directory);
}
